scrape/scrape.go

```diff
@@ -589,6 +606,8 @@ type cacheEntry struct {
 }
 
 type scrapeLoop struct {
+	jobName string
```

Comment:
I think I'd make the jobName a field on the scrapePool instead of on every scrapeLoop.

Reply:
Yeah, I tried that. The problem with it is that the `scrapeLoop` has no notion of the `scrapePool` and I need the name within the `scrapeCache` (with is part of the loop). Originally, I even tried passing the `jobName` to the `scrapeCache` but it feels the cache should have no notion of a "job".